twisted/conch/insults/insults.py

```diff
@@ -689,196 +694,200 @@ def tilde(self, proto, handler, buf):
     def cursorUp(self, n=1):
         assert n >= 1
         self.cursorPos.y = max(self.cursorPos.y - n, 0)
-        self.write('\x1b[%dA' % (n,))
+        self.write(b'\x1b[%dA' % (n,))
 
 
     def cursorDown(self, n=1):
         assert n >= 1
         self.cursorPos.y = min(self.cursorPos.y + n, self.termSize.y - 1)
-        self.write('\x1b[%dB' % (n,))
+        self.write(b'\x1b[%dB' % (n,))
 
 
     def cursorForward(self, n=1):
         assert n >= 1
         self.cursorPos.x = min(self.cursorPos.x + n, self.termSize.x - 1)
-        self.write('\x1b[%dC' % (n,))
+        self.write(b'\x1b[%dC' % (n,))
 
 
     def cursorBackward(self, n=1):
         assert n >= 1
         self.cursorPos.x = max(self.cursorPos.x - n, 0)
-        self.write('\x1b[%dD' % (n,))
+        self.write(b'\x1b[%dD' % (n,))
 
 
     def cursorPosition(self, column, line):
-        self.write('\x1b[%d;%dH' % (line + 1, column + 1))
+        self.write(b'\x1b[%d;%dH' % (line + 1, column + 1))
 
 
     def cursorHome(self):
         self.cursorPos.x = self.cursorPos.y = 0
-        self.write('\x1b[H')
+        self.write(b'\x1b[H')
 
 
     def index(self):
         self.cursorPos.y = min(self.cursorPos.y + 1, self.termSize.y - 1)
-        self.write('\x1bD')
+        self.write(b'\x1bD')
 
 
     def reverseIndex(self):
         self.cursorPos.y = max(self.cursorPos.y - 1, 0)
-        self.write('\x1bM')
+        self.write(b'\x1bM')
 
 
     def nextLine(self):
         self.cursorPos.x = 0
         self.cursorPos.y = min(self.cursorPos.y + 1, self.termSize.y - 1)
-        self.write('\n')
+        self.write(b'\n')
 
 
     def saveCursor(self):
         self._savedCursorPos = Vector(self.cursorPos.x, self.cursorPos.y)
-        self.write('\x1b7')
+        self.write(b'\x1b7')
 
 
     def restoreCursor(self):
         self.cursorPos = self._savedCursorPos
         del self._savedCursorPos
-        self.write('\x1b8')
+        self.write(b'\x1b8')
 
 
     def setModes(self, modes):
         # XXX Support ANSI-Compatible private modes
-        self.write('\x1b[%sh' % (';'.join(map(str, modes)),))
+        self.write(b'\x1b[%sh' %
+                   (b';'.join([intToBytes(mode) for mode in modes])))
 
```

Comment:
Yes, you are right on this point.
I want to come back and fix these places for Python 3.3 and Python 3.4 in
a follow-on ticket.  Right now I'd like to push this patch set in, because it is
a bit easier to review a smaller subset of work, and all the unit tests still work on Python 2.7.

When the conch modules are fully ported to Python 3.3, 3.4, 3.5, then I will add those
modules to twisted/python/dist3.py, so that these modules will be tested on the Python 3.3, 3.4, 3.5 Travis environments.  If the modules are not in dist3.py, they will be only tested on Python 2.7.